app/components/Sidebar/Sidebar.tsx

```diff
@@ -145,41 +147,43 @@ const Sidebar = React.forwardRef<HTMLDivElement, Props>(
 
     return (
       <>
-        <Container
-          ref={ref}
-          style={style}
-          $isAnimating={isAnimating}
-          $isSmallerThanMinimum={isSmallerThanMinimum}
-          $mobileSidebarVisible={ui.mobileSidebarVisible}
-          $collapsed={collapsed}
-          column
-        >
-          {ui.mobileSidebarVisible && (
-            <Portal>
-              <Backdrop onClick={ui.toggleMobileSidebar} />
-            </Portal>
-          )}
-          {children}
-          <ResizeBorder
-            onMouseDown={handleMouseDown}
-            onDoubleClick={ui.sidebarCollapsed ? undefined : handleReset}
-          />
-          {ui.sidebarCollapsed && !ui.isEditing && (
+        <SidebarContext.Provider value={setOpenMenus}>
+          <Container
+            ref={ref}
+            style={style}
+            $isAnimating={isAnimating}
+            $isSmallerThanMinimum={isSmallerThanMinimum}
+            $mobileSidebarVisible={ui.mobileSidebarVisible}
+            $collapsed={collapsed}
+            column
+          >
+            {ui.mobileSidebarVisible && (
+              <Portal>
+                <Backdrop onClick={ui.toggleMobileSidebar} />
+              </Portal>
+            )}
+            {children}
+            <ResizeBorder
+              onMouseDown={handleMouseDown}
+              onDoubleClick={ui.sidebarCollapsed ? undefined : handleReset}
+            />
+            {ui.sidebarCollapsed && !ui.isEditing && (
+              <Toggle
+                onClick={ui.toggleCollapsedSidebar}
+                direction={"right"}
+                aria-label={t("Expand")}
+              />
+            )}
+          </Container>
+          {!ui.isEditing && (
             <Toggle
+              style={toggleStyle}
               onClick={ui.toggleCollapsedSidebar}
-              direction={"right"}
-              aria-label={t("Expand")}
+              direction={ui.sidebarCollapsed ? "right" : "left"}
+              aria-label={ui.sidebarCollapsed ? t("Expand") : t("Collapse")}
             />
           )}
-        </Container>
-        {!ui.isEditing && (
```

Comment:
I feel like this was outside of the container for a very good reason because the refactor is too obvious, but for the life of me I'm not sure what.